network-grpc/src/client.rs

```diff
@@ -69,12 +72,12 @@ pub mod chain_bounds {
 ///
 /// This type encapsulates the gRPC protocol client that can
 /// make connections and perform requests towards other blockchain nodes.
-pub struct Client<B, S, E> {
+pub struct Client<B, T, S, E> {
     node: gen_client::Node<Connection<S, E, BoxBody>>,
-    _phantom: PhantomData<B>,
+    _phantom: PhantomData<(B, T)>,
 }
 
-impl<B, S, E> Client<B, S, E>
+impl<T, G, S, E> Client<T, G, S, E>
```

Comment:
Please use the same names for type parameters of `Client` throughout the source file. It's very confusing to have `T` represent different parameters in different places.